parser-typechecker/src/Unison/Typechecker/Components.hs

```diff
@@ -35,10 +35,10 @@ ordered = ABT.orderedComponents
 --
 -- Fails on the left if there are duplicate definitions.
 minimize ::
-  Var v =>
+  (Var v) =>
   Term' vt v a ->
   Either (NonEmpty (v, [a])) (Maybe (Term' vt v a))
-minimize (Term.LetRecNamedAnnotatedTop' isTop ann bs e) =
+minimize (Term.LetRecNamedAnnotatedTop' isTop blockAnn bs e) =
```

Comment:
Minimization re-arranges and re-writes let-rec bindings;

The previous version incorrectly uses the _start_ of each binding block as the annotation for the whole block, which was mostly fine because the Abs annotation rarely shows up in error messages, but it breaks our ability to find the node which corresponds to a given file position.

Now, we simply ascribe the whole block to each Abs node, which isn't perfect, but IMO is at least AS correct as the previous behaviour and might actually behave better in cases where binding orders get re-arranged. Crucially, it means all abt node spans properly nest within one another.